mava/utils/total_timestep_checker.py

```diff
@@ -20,23 +20,29 @@
 
 def check_total_timesteps(config: DictConfig) -> DictConfig:
     """Check if total_timesteps is set, if not, set it based on the other parameters"""
-    n_devices = len(jax.devices())
+
+    if config.arch.architecture_name == "anakin":
+        n_devices = len(jax.devices())
+        update_batch_size = config.system.update_batch_size
+    else:
+        n_devices = 1  # We only use a single device's output when updating.
```

Comment:
Not sure if this is correct, because firstly we need to decide is timestep calculated based on learner updates_timesteps or based on the actor timesteps, besides the n_devices can be more than one in both options as we can decide the actor and the learner devices number

Reply:
I based the total timesteps on the learner, and I think it should stay that way. The actors can take steps that get discarded if the pipeline is full, so it's better to track the total number of training samples we're going to update over which is (currently)  the output of a single actor (num_envs * rollout_length) * num_updates.

Regarding n_devices, although we have multiple learner devices, we only have a single learner thread. This thread splits what it receives from an actor across the learner devices.

Reply:
Good point, we can't pmap this because it will never be a jittable env and we're only doing eval on 1 device.